javascript/javascript3/week3/shoppingcart.js

```diff
@@ -0,0 +1,104 @@
+const productUl = document.getElementById("products");
+const totalPrice = document.getElementById("total");
+const username = document.getElementById("user");
+const productPrice = document.getElementById("price");
+
+class Product {
+    constructor(name, price) {
+        this.name = name;
+        this.price = price;
+    }
+
+    convertToCurrencyWithApi(currency) {
+        const apiUrl = `https://api.exchangeratesapi.io/latest?base=DKK`;
+        return fetch(apiUrl)
+            .then(response => response.json())
+            .then(data => {
+                return this.price * data.rates[currency]
+
+            })
+            .catch(err => console.log(err));
+    }
+}
+
+class ShoppingCart {
+    constructor() {
+        this.products = [];
+    }
+
+    addProduct(product) {
+        this.products.push(product);
+    }
+
+    removeProduct(product) {
+        const newProducts = this.products.filter((existingProduct) => {
+            return existingProduct.name !== product.name
+        })
+        this.products = newProducts;
+    }
+
+    searchProduct(productName) {
+        const searchedProduct = this.products.filter(product => product.name.toLowerCase() === productName.toLowerCase());
+        return searchedProduct;
+    }
+
+    renderProducts() {
+        this.products.forEach((product) => {
+            const li = document.createElement('li');
+            const liPrice = document.createElement("li");
+            li.textContent = `${product.name}`;
+            liPrice.textContent = `${product.price}`;
+            productUl.appendChild(li);
+            productPrice.appendChild(liPrice);
+        })
+        return products;
+    }
+
+    getTotal() {
+        const totalAmount = this.products.map(product => product.price).reduce((acc, currentvalue) => acc + currentvalue);
+        totalPrice.textContent = `The total amount is: ${totalAmount} kr`;
+        console.log(totalAmount);
+    }
+
+    getUser() {
```

Comment:
This method was supposed to accept a user as a parameter. You can fetch for all users and then filter the requested one.